### PR TITLE
Updates project to use yarn and dependencies

### PR DESCRIPTION
**Pull request by mateus**
This PR updates the project to use `yarn` instead of `npm` and updates the dependencies to their latest versions.

Considering the following updates:
- Removed `package-lock.json`
- Running `yarn install` and `yarn ugrade --latest`

### CODE REVIEW

File: resources/webview.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
           'appName': 'marketing-misc',
           'defaultAttributes': {
             'plugin-identifier': 'polaris-telescope',
-            'polarisInSketchVersion': '0.9.4'
+            'polarisInSketchVersion': '0.9.5'
```
Comment (kaelig) [inline resources/webview.html]: Is this bump intentional?

Comment (mateus) [reply resources/webview.html]: Yeap. The release script doesn't update this file automatically, we have to do it manually.

**Comment by kaelig** [comment]
Several dependencies weren't updated, is that on purpose?

screenshot 2018-05-07 15 36 41

File: resources/webview.html (hunk 1 above)
Comment (kaelig) [reply resources/webview.html]: Ok - not sure this should be done in this PR then

**Comment by kaelig** [comment]
Can you also update the contribution guidelines to use yarn? https://github.com/Shopify/polaris-telescope/blob/master/CONTRIBUTING.md#how-to-run-locally

**Comment by mateus** [comment]
The webview update is a breaking change. I will address that on a follow up PR.

**Comment by kaelig** [comment]
Ping!

> Can you also update the contribution guidelines to use yarn? https://github.com/Shopify/polaris-telescope/blob/master/CONTRIBUTING.md#how-to-run-locally

**Comment by kaelig** [comment]
I took the liberty of picking this PR up and update more dependencies, as well as get some tooling / linting in there. Next step: updating the webview!

**Pull request merged**